src/main/resources/db/changesets/20231129_snapshotrequests.yaml

```diff
@@ -0,0 +1,60 @@
+databaseChangeLog:
+  - changeSet:
+      id: snapshot_requests
+      author: rjohanek
+      changes:
+        - createTable:
+            tableName: snapshot_requests
+            columns:
+              - column:
+                  name: id
+                  type: ${uuid_type}
+                  defaultValueComputed: ${uuid_function}
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+              - column:
+                  name: dataset_id
+                  type: ${uuid_type}
+                  constraints:
+                    nullable: false
+                    foreignKeyName: fk_dataset_snapshot_requests
+                    references: dataset(id)
+                    deleteCascade: true
+              - column:
+                  name: request_name
+                  type: text
+                  constraints:
+                    nullable: false
+              - column:
+                  name: request_research_purpose
```

Comment:
Same type of thinking here, maybe this is `research_purpose`

Reply:
or snapshot_research_purpose? for consistency with snapshot_name

Reply:
Maybe? The snapshot_name will be used when creating the snapshot. The research purpose is for the request alone, and isn't applied to the new snapshot. So they seem slightly different to me.

Reply:
Why is the research purpose not for the snapshot as well? Since they are collected on the same form I thought they were both applied to the snapshot